### ISSUE TITLE
⚒️ Set up testing/linting infrastructure/workflow

### ISSUE THREAD

**Issue by tatianamac** [tatianamac/selfdefined#74]
Ensure considers accessibility per #48.

@ovlb Could you please add any considerations here since you have a stronger sense of what should be done here?

**Comment by ovlb** [comment]
Sure. This list is considering future developments of the site, so some things might seem overkill right now. They will save us from grinding teeth and loss of hair later, though. 

## Tools to be integrated

- [ ] ESLint
- [x] Prettier
- [ ] Stylelint
- [x] Editorconfig
- [ ] axe
- [ ] Unit Testing (personal preference is Jest or ava)
- [ ] Codecov (or a similar tool for collecting test coverage)
- [ ] e2e testing (Cypress?)

## Dev infrastructure

These tooling will enable us to ensure consistent code quality across branches and remove the need for manual style corrections in PRs. 

If we find a way to apply the formatting configured through the tools above for contributors who are not able to set up an advanced dev workflow on their machines (no shame in that) this will be A++.

- [ ] Husky or another solution for running npm scripts pre-push and/or -commit
- [ ] lint-staged
- [ ] CI (CircleCI or GitHub actions might make sense; CD is handled by Netlify)



There are probably things I miss right now. Will keep the list updated. And if anyone reading this has input, please post it!

This list shall act as a tracking issue.

If you want to work on integrating of one or more of these tools, please open a PR and I will link it in the list. Thanks.

**Comment by greatislander** [comment]
@tatianamac @ovlb I've been working with husky/lint-staged/etc a lot lately to handle linting on commit, as well as setting up corresponding GitHub CI. Would be happy to start on this.

**Comment by ovlb** [comment]
@greatislander Please go ahead :) Shall I add a basic ESLint config, so that your workflow has something to test against or would you do this too?

**Comment by greatislander** [comment]
That would be great, @olvb!

**Comment by ovlb** [comment]
@greatislander I’ve opened #78 for linting and testing. Once this is merged you can go ahead with the CI :)

**Comment by codeAbul** [comment]
@ovlb Once CI system is set up, I can work on setting up axe!

**Comment by ovlb** [comment]
> @ovlb Once CI system is set up, I can work on setting up axe!

Great!

**Comment by ovlb** [comment]
@greatislander #78 is merged. :)

**Comment by greatislander** [comment]
@ovlb @tatianamac I am happy to scaffold a Stylelint configuration as well. Do you have any preferences? I typically use the recommended rules with [stylelint-order](https://github.com/hudochenkov/stylelint-order) and [`properties-alphabetical-order`](https://github.com/hudochenkov/stylelint-order/blob/master/rules/properties-alphabetical-order/README.md) set to `true` but let me know your thoughts!

**Comment by tatianamac** [comment]
Thanks for your patience—just flew back from Australia. I am good with those. @ovlb, what're your preferences?

**Comment by ovlb** [comment]
@codeAbul Do you still want to set up axe? CI is ready now.

**Comment by tatianamac** [comment]
Also, I imagine we should add some notes to our documentation? I can add it as a separate issue if we agree (once we're done).

**Comment by greatislander** [comment]
@tatianamac That makes sense! Happy to add documentation for the configuration I added.

**Comment by ovlb** [comment]
Much appreciated. Maybe by expanding this document? https://github.com/tatianamac/selfdefined/blob/master/11ty/documentation/development.md

**Comment by tatianamac** [comment]
As @codeAbul needs to step away from the aXe portion, @greatislander or @ovlb are either one of you able to take on setting that up? Otherwise I'm happy to recruit other folks if you're not able.

**Comment by greatislander** [comment]
@tatianamac I can take a crack at it.

**Comment by tatianamac** [comment]
@greatislander Did we lose track of this issue? 😬 (I know you're working hard on the Open Authoring one, so no pressure from me—just trying to get through the issues 😅)

**Comment by greatislander** [comment]
Oh yeah I totally lost track of it. Oops. I am going to be setting up axe for a work project so once I've figured it out there I will replicate here :grin:

**Comment by tatianamac** [comment]
@greatislander Did you figure out the axe set up? I'd love to add it here if you did! 🤓🙏🏽

**Comment by greatislander** [comment]
I spent a bit of time on it last week but ran into some issues with GitHub Actions which I haven't figured out yet (got it working locally but couldn't get it to run in CI). Still chipping away at it, and will probably spend some more time this week. Fingers crossed!
